openedx/core/djangoapps/credit/utils.py

```diff
@@ -0,0 +1,35 @@
+"""
+Utilities for the credit app.
+"""
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+
+
+def get_course_blocks(course_key, category):
+    """
+    Retrieve all XBlocks in the course for a particular category.
+
+    Returns only XBlocks that are published and haven't been deleted.
+    """
+    return [
+        block for block in modulestore().get_items(
+            course_key,
+            qualifiers={"category": category},
+            revision=ModuleStoreEnum.RevisionOption.published_only,
+        )
+        if _is_in_course_tree(block)
+    ]
```

Comment:
This generic utility function should be somewhere in the modulestore lib so that other code can use it too.